### DISCUSSION TITLE
Get version number of RAUC update bundle without using RAUC

### DISCUSSION THREAD

**Discussion by stefanie-mak-lucas** [rauc/rauc#1447] [Q&A] [answered]
Hello, we would like to enable our customers to update the firmware on their devices via a connected PC. The customers do not have direct access to the firmware, only via our client application on the PC.

Currently, we would imagine the workflow as follows: Our customers download the most recent RAUC update bundle from our website, and then transfer the bundle to the device via the client software on the connected PC. Then we would either detect the new file and start installing the update automatically, or we would communicate with RAUC via the DBUS API and tell it to start the update.

Ideally, we would want our client application to show the version of the update before the file transfer. However, RAUC is not installed on the PC, only on the firmware on our device. Can you think of any way how our client software could find out the version number of the update bundle before the file transfer, that would not require us to install RAUC on the customer PC?

**Comment by jluebbe** [comment, marked as answer]
It sounds like that in your case, the device doesn't have Internet access itself and so can't download or stream the bundle directly?

For bundles in the verity format (which we'd strongly recommend for all current projects), the manifest is contained in the CMS signature at the end of the bundle: https://rauc.readthedocs.io/en/latest/reference.html#verity-format. You can easily extract that data. To show the necessary steps, I've just created a PR: https://github.com/rauc/rauc/pull/1451.

Note that you need to check the CMS signature before trusting the contents of the manifest.

If you want to stream the bundle from the client application to the device, you can take a look at this discussion: https://github.com/rauc/rauc/discussions/1440#discussioncomment-9675936.

**Comment by stefanie-mak-lucas** [comment]
Thanks a lot, this is great news!
